test/models/watching_transaction_test.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class WatchingTransactionTest < ActiveSupport::TestCase
+  test 'create new watching transaction should work' do
+    params = attributes_for(:watching_transaction)
+
+    tx = WatchingTransaction.new(params)
+
+    user = create(:user)
+    tx.user_id = user.id
+
+    assert tx.save!,
+           'should save'
+
+    assert_not WatchingTransaction.new(params).save,
+               'should fail with the same data'
+  end
+
+  test 'watch should work' do
+    user = create(:user)
+    attrs = attributes_for(:watching_transaction)
+    ok, tx = WatchingTransaction.watch(user, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+    assert_equal user.id, tx.user.id,
+                 'should have correct user id'
+  end
+
+  test 'resend should work' do
+    user = create(:user)
+    watching_transaction = create(:watching_transaction, user: user)
+    attrs = attributes_for(:watching_transaction)
+
+    ok, tx = WatchingTransaction.resend(user, watching_transaction, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+    assert_equal watching_transaction.group_id, tx.group_id,
+                 'should have the same group as the previous transaction'
+  end
+
+  test 'resend transactions should keep all if none is mined' do
+    original = FactoryBot.create(:watching_transaction)
+    4.times do
+      FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+    end
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+                  .with(body: /eth_getTransactionByHash/)
+                  .to_return(
+                    body: {
+                      result: {
+                        'block_number' => nil,
+                      }
+                    }.to_json
+                  )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 5)
+    assert_equal 5, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should work'
+  end
+
+  test 'resend transactions should destroy all if one is mined' do
+    original = FactoryBot.create(:watching_transaction)
```

Comment:
Same comment as above with creating a random group of transactions. Here you also need to remove or let go of `original` and `latest` and rely on the transaction group generated.

Something like this:

```
group_id = SecureRandom.uuid
size = SecureRandom.rand(1..7)
user = FactoryBot.create(:user)

transaction_group = (1..size).map do
  FactoryBot.create(:watching_transaction, group_id: group_id, user: user)
end
```